cluster/gce/gci/configure-helper.sh

```diff
@@ -2578,6 +2578,10 @@ EOF
 function override-kubectl {
     echo "overriding kubectl"
     echo "export PATH=${KUBE_HOME}/bin:\$PATH" > /etc/profile.d/kube_env.sh
+    # Add ${KUBE_HOME}/bin into sudoer secure path.
+    SUDO_PATH="$(sudo env | grep "^PATH=")"
```

Comment:
Use `declare` to make this variable local. You can also use `sed` to both strip the `"^PATH="` prefix and print the output:

I also recommend removing the outer double quotes, to avoid having nested quotes in your expression. They're not really needed when you have a variable assignment.

Also, you should handle the case where what you get is empty. (That case doesn't make much sense, but the current handling will add the current directory to $PATH, which is really bad...)

```
    declare sudo_path
    sudo_path=$(sudo env | sed -ne 's/^PATH=//p')
    echo "Defaults secure_path=\"${KUBE_HOME}/bin${sudo_path:+:${sudo_path}}\"" >/etc/sudoers.d/kube_secure_path
```

Reply:
Ah and those files need to be created without world-writable permissions, so:

```shell
    declare sudo_path
    sudo_path=$(sudo env | sed -ne 's/^PATH=//p')
    (
        umask 027
        echo "Defaults secure_path=\"${KUBE_HOME}/bin${sudo_path:+:${sudo_path}}\"" >/etc/sudoers.d/kube_secure_path
    )
```

Reply:
> I also recommend removing the outer double quotes, to avoid having nested quotes in your expression.

Always quoting string seems to be some google convention. Anyway, removed it.

Reply:
Done.

Reply:
> Always quoting string seems to be some google convention. Anyway, removed it.

Yes, it's in the shell style guide... But mainly because it's easier to explain than single out the cases when it's not needed...

Variable assignments don't need it (since they don't do word splitting), so not using them there is generally a great way to prevent nested double quotes, which are really unhelpful on vulnerabilities...